internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/endToEnd/TestServerStorePropertiesEndpoint.java

```diff
@@ -0,0 +1,96 @@
+package com.linkedin.venice.endToEnd;
+
+import static org.testng.Assert.*;
+
+import com.linkedin.venice.client.store.ClientConfig;
+import com.linkedin.venice.client.store.ClientFactory;
+import com.linkedin.venice.client.store.transport.TransportClient;
+import com.linkedin.venice.client.store.transport.TransportClientResponse;
+import com.linkedin.venice.controllerapi.UpdateStoreQueryParams;
+import com.linkedin.venice.fastclient.utils.AbstractClientEndToEndSetup;
+import com.linkedin.venice.integration.utils.VeniceServerWrapper;
+import com.linkedin.venice.meta.QueryAction;
+import com.linkedin.venice.meta.Store;
+import com.linkedin.venice.metadata.response.StorePropertiesResponseRecord;
+import com.linkedin.venice.security.SSLFactory;
+import com.linkedin.venice.serializer.FastSerializerDeserializerFactory;
+import com.linkedin.venice.serializer.RecordDeserializer;
+import com.linkedin.venice.utils.SslUtils;
+import com.linkedin.venice.utils.TestUtils;
+import java.util.Optional;
+import org.apache.avro.Schema;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+
+public class TestServerStorePropertiesEndpoint extends AbstractClientEndToEndSetup {
+
+  /*
+   * @{link AbstractClientEndToEndSetup} enables the read quota for stores as part of prepareData and stores are
+   * typically reused at a class level for tests. However, some tests require flows to disable read quota, and
+   * it is important to reset so that these tests don't step on other tests. For now, we choose to add a before
+   * method to class that validates disabled read quota behavior to reset the state to ensure other tests run
+   * successfully.
+   */
+  @BeforeMethod
+  public void enableStoreReadQuota() {
```

Comment:
It was copied from the old test file. If its not needed, will remove

Reply:
See the later comments, I think it's needed because `checkStore` inherited some FC gatekeeping logic to reject and return 403 if SN read quota is not enabled. We should change that.